### DISCUSSION TITLE
http.response.status_code Activitiy not accessible on Processor?

### DISCUSSION THREAD

**Discussion by renatosc** [open-telemetry/opentelemetry-dotnet#5395] [General]
I created a custom Processor and add it as the last one before my ConsoleExporter. I see on my console that my request does have an Activity.Tag "http.response.status_code" but my Processor doesn't not see that tag, which means it is being defined after my Processor. But as I said, my Processor is the last one added. So, does that mean that I cannot access that Activity tag using a Processor? Are there any other alternatives to have access to that http.response.status_code tag?

**Comment by cijothomas** [comment]
Can you share a minimal repro? Also, make sure you are inspecting the Activity in the onend callback, not onstart in the processor.

**Reply by renatosc** [reply]
Yes, I am doing it on the onEnd.

```
public class MyProcessor : BaseProcessor<Activitiy> 
{
    public override void OnEnd(Activity activity)
   {
        // here I search the activity.Tags for the "http.response.status_code" but there is none. I do see other tags like url.path
   }
}
```
and I have registered MyProcessor in my Program.cs as in:

```
builder.Services.AddOpenTelemetry().WithTracing( options => {
    options
        .AddHttpClientInstrumentation()
        .AddAspNectCoreInstrumentation()
        .AddProcessor(new MyProcessor())
        .AddOtlpExporter()
        .AddConsoleExporter()
})
```

**Reply by renatosc** [reply]
When I get back at computer I will create a mini complete project, upload to github and share it here

**Reply by renatosc** [reply]
Here is an example (.NET 8 Web Api):

Program.cs:
```
using OpenTelemetry;
using OpenTelemetry.Trace;
using System.Diagnostics;

var builder = WebApplication.CreateBuilder(args);

builder.Services
    .AddOpenTelemetry()
    .WithTracing(options =>
    {
        options
            .AddHttpClientInstrumentation()
            .AddAspNetCoreInstrumentation()
            .AddProcessor(new MyCustomProcessor())
            .AddConsoleExporter();
    });

var app = builder.Build();
app.MapGet("/weatherforecast", () => { return "Hello World"; });
app.Run();

class MyCustomProcessor: BaseProcessor<Activity>
{
    public override void OnEnd(Activity data)
    {
        string? httpStatusCode = null;
        foreach (var kvp in data.Tags)
        {
            if (kvp.Equals("http.response.status_code")) {
                httpStatusCode = kvp.Value;
                Console.WriteLine("FOUND IT!");
            }
        }
        if (httpStatusCode == null) {
            Console.WriteLine("NOT found it!");
        }
    }
}
```

Here is a link for the entire solution: https://github.com/renatosc/otelcustomprocessor

<img width="1253" alt="img" src="https://github.com/open-telemetry/opentelemetry-dotnet/assets/6549539/cd5431dd-78ad-45dd-8266-a4df84c4e60b">

**Reply by renatosc** [reply]
@cijothomas do you see anything wrong in the code above or should I file a bug?

**Reply by vishweshbankwar** [reply]
> kvp.Equals("http.response.status_code")

use .key

**Reply by vishweshbankwar** [reply]
@renatosc I re-checked this. I am able to repro it. Its odd. Will take a look. Do you mind opening an issue https://github.com/open-telemetry/opentelemetry-dotnet/issues

**Reply by vishweshbankwar** [reply]
never mind - You are looking at Tags. You should look at TagObjects
See https://github.com/open-telemetry/opentelemetry-dotnet/blob/42ecd73bd0956ae41e67894b93d1e3a89a4bbba8/src/OpenTelemetry.Exporter.Console/ConsoleActivityExporter.cs#L45

**Reply by renatosc** [reply]
Thank you. That was it. I read about TagObjects but I thought that it was only related to custom exporters.

**Reply by cijothomas** [reply]
https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/docs/trace/extending-the-sdk#enriching-processor It'd be good to update these docs to call this out. A common mistake I guess to confuse between Tags and TagObjects, in the absence of docs.

**Reply by renatosc** [reply]
@cijothomas  I submitted a PR to update the docs as you suggested. The Exporters section already have that this called out, but the Processors one didn't, so I added one there as well.

https://github.com/open-telemetry/opentelemetry-dotnet/pull/5445

Thank you again for the help.
